src/HonzaBotner.Discord.Services/EventHandlers/StaffVerificationEventHandler.cs

```diff
@@ -1,48 +1,41 @@
 ﻿using System.Linq;
 using System.Threading.Tasks;
+using DSharpPlus;
 using DSharpPlus.Entities;
 using DSharpPlus.EventArgs;
 using HonzaBotner.Discord.EventHandler;
-using HonzaBotner.Discord.Services.Options;
 using HonzaBotner.Services.Contract;
 using HonzaBotner.Services.Contract.Dto;
 using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
 
 namespace HonzaBotner.Discord.Services.EventHandlers
 {
-    public class StaffVerificationEventHandler : IEventHandler<MessageReactionAddEventArgs>,
-        IEventHandler<MessageReactionRemoveEventArgs>
+    public class StaffVerificationEventHandler : IEventHandler<ComponentInteractionCreateEventArgs>
     {
         private readonly IUrlProvider _urlProvider;
-        private readonly CommonCommandOptions _config;
         private readonly DiscordRoleConfig _discordRoleConfig;
         private readonly IDiscordRoleManager _roleManager;
         private readonly ILogger<StaffVerificationEventHandler> _logger;
 
         public StaffVerificationEventHandler(IUrlProvider urlProvider,
-            IOptions<CommonCommandOptions> options,
             IOptions<DiscordRoleConfig> discordRoleConfig,
             IDiscordRoleManager roleManager,
             ILogger<StaffVerificationEventHandler> logger)
         {
             _urlProvider = urlProvider;
-            _config = options.Value;
             _discordRoleConfig = discordRoleConfig.Value;
             _roleManager = roleManager;
             _logger = logger;
         }
 
-        public async Task<EventHandlerResult> Handle(MessageReactionAddEventArgs eventArgs)
+        public async Task<EventHandlerResult> Handle(ComponentInteractionCreateEventArgs eventArgs)
         {
-            if (!(eventArgs.Message.Id == _config.VerificationMessageId
-                  && eventArgs.Message.ChannelId == _config.VerificationChannelId))
-                return EventHandlerResult.Continue;
-            if (!eventArgs.Emoji.Name.Equals(_config.StaffVerificationEmojiName)) return EventHandlerResult.Continue;
+            if (eventArgs.Id != "staff-verification") return EventHandlerResult.Continue;
```

Comment:
-> move to config

Reply:
I disagree here. Button's IDs are internal variables which should be unique at all costs and making them "openly" available for change just makes more space for errors than it fixes. In the future, if we introduce buttons generaten on the go, we will need to think of some way to generate unique IDs. It's easier to know that we have 2 special IDs we can't use, than to risk that user will insert his own in config and it might one day conflict.

Reply:
Probably yes, but hardcoded constants are not the way to go. @ostorc opinion?

Reply:
Moved to config. Strongly discourage from editing it tho!